File: 713. Subarray Product Less Than K.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int left = 0;
        int product = 1;
        int answer = 0;
        for (int right = 0; right < nums.length; right++) {
            product *= nums[right];
            while (left <= right && product >= k) {
                product /= nums[left++];
            }
            answer += right - left + 1;
        }
        return answer;
    }
}

File: 713. Subarray Product Less Than K.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        current, left, right = 1, 0, 0
        answer = 0
        while right < len(nums):
            current *= nums[right]
            while current >= k and left <= right:
                current /= nums[left]
                left += 1
            answer += right - left + 1
            right += 1
        return answer

